packages/rest/src/rest.server.ts

```diff
@@ -58,6 +58,10 @@ export interface HttpServerLike {
   requestHandler: HttpRequestListener;
 }
 
+interface ExpressRouter extends express.Router {
```

Comment:
`ExpressRouter` is used in `rest.server.ts` and `routing-table.ts`, and defined in both the files. What is a good location to export it from to avoid duplication?

Having to extend `express.Router` because its definition file does not include the `handle` method for it. I wonder why.

Reply:
Please note that `express.Router` itself is  the `handle` function.

Reply:
`src/types.ts` is a good place to define common interfaces

Reply:
As @raymondfeng said, `express.Router` is a callable function. It extends `IRouter` which extends `RequestHandler` which is defined as follows:

```ts
export interface RequestHandler {
    // tslint:disable-next-line callable-types (This is extended from and can't extend from a type alias in ts<2.2
    (req: Request, res: Response, next: NextFunction): any;
}
```